### PR TITLE
fix bidirectional streaming for grpc

### PR DESCRIPTION
**Pull request by sredxny**
## Description
Fix grpc bidirectional streaming functionality

## Related Issue
https://tyktech.atlassian.net/browse/TT-2693

## Motivation and Context
fix https://tyktech.atlassian.net/browse/TT-2693

## How This Has Been Tested
The same as reported in the ticket

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: gateway/reverse_proxy.go [hunk 1]
```diff
@@ -1478,13 +1518,21 @@ func copyBody(body io.ReadCloser) io.ReadCloser {
 }
 
 func copyRequest(r *http.Request) *http.Request {
+	if r.ContentLength == -1 {
```
Comment (buger) [inline gateway/reverse_proxy.go:1521]: Add comments here, saying that this line for the case when request/response body is streaming.

Comment (sredxny) [reply gateway/reverse_proxy.go:1521]: done

**Comment by buger** [comment]
Any ideas why tests fail?

**Comment by sredxny** [comment]
@buger something with goreleaser, I go this message:

```
 release failed after 202.34s error=invalid nfpm config: cannot add coprocess/bindings/python/coprocess_return_overrides_pb2.py because coprocess/python/proto/coprocess_return_overrides_pb2.py is already present at the same destination (/opt/tyk-gateway/coprocess/python/proto/coprocess_return_overrides_pb2.py): content collision
Error: Process completed with exit code 1.
```
Not sure what does that means 👀  but the unit test are passing

**Comment by buger** [comment]
Do not think it is related to goreleaser, but I have no idea why this GH comment about failed test is there.

**Comment by sredxny** [comment]
At this point all the tests passed, but I noticed some unestanility on `TestReloadLoop` tests

**Pull request merged**